internal/bot/bot.go

```diff
@@ -51,64 +55,67 @@ func main() {
 	if err != nil {
 		log.Error("Failed connect to database.")
 		log.Error(err)
-		os.Exit(2)
+		os.Exit(1)
 	}
 	log.Info("Successfully connected to database.")
 
 	bot, err := tg.NewBotAPI(cfg.BotToken)
 	if err != nil {
-		log.Error("Failed connect to Telegram.")
-		log.Error(err)
-		os.Exit(3)
+		log.Errorf("Failed connect to Telegram. %v", err)
+		os.Exit(1)
 	}
+	log.Infof("Authorized on telegram account @%s", bot.Self.UserName)
+
 	bot.Debug = cfg.TelegramDebug
 
-	app := App{
+	return &Bot{
 		Config: cfg,
 		DB:     db,
-		Bot:    bot,
+		API:    bot,
+		Log:    log,
 	}
+}
 
-	log.Infof("Authorized on telegram account @%s", bot.Self.UserName)
-
+// Start starts polling for messages for bot
+func (b *Bot) Start() {
 	u := tg.NewUpdate(0)
 	u.Timeout = 60
 
-	updates, err := bot.GetUpdatesChan(u)
+	updates, err := b.API.GetUpdatesChan(u)
+
+	if err != nil {
+		b.Log.Errorf("Failed get updates from Telegram. %v", err)
+		os.Exit(1)
+	}
 
 	for update := range updates {
 		// Create context for request
 		reqTime := time.Now()
-		ctx := Ctx{
-			App:         app,
-			Log:         log,
-			RequestID:   reqTime.UnixNano() / 1000,
-			RequestTime: reqTime,
+		req := BotRequest{
+			ID:   reqTime.UnixNano() / 1000,
+			Time: reqTime,
 		}
 
+		ctx := context.WithValue(context.Background(), botReqKey, req)
+
 		switch {
 		case update.Message.IsCommand():
-			log.WithFields(logrus.Fields{
-				"requestID": ctx.RequestID,
+			b.Log.WithFields(logrus.Fields{
+				"requestID": req.ID,
 				"user":      update.Message.From,
 			}).Infof("Command request %s %s", update.Message.Command(), update.Message.CommandArguments())
-			go command(ctx, update.Message)
+			go b.HandleCommand(ctx, update.Message)
```

Comment:
looks like you can didn't use ctx here, you can just put request as argument. And use ctx for deadline/cancelation request.

Reply:
Telegram Bot API don't cancel requests. I not need in deadline and cancelation here.